### PR TITLE
DD4hep: Add 2021 CMS Geometry Scenario

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* Define a complete 2021 scenario and make it a default one
* Make sure all materials are defined and are in a required order
   * the XML to DD4hep converter requires that a composite material refers to already defined materials
* Replace a ```trackermaterial:T_Copper``` with an identical ```materials:Copper```
* Replace a ```trackermaterial:T_Alluminium``` with an identical ```materials:Alluminium```
* Add a ```cms::DDG4Builder``` to fill in the sensitive detector catalog
* Print the sensitive detector catalog

@cvuosalo and @vargasa - FYI

```
 cmsRun SimG4Core/DD4hepGeometry/test/python/testZMM_13TeV_TuneCUETP8M1_cfi_GEN_SIM.py
```
will run and create a G4 geometry, the G4 production cuts and the sensitive dets catalog, but it will fail on finding an overlap:
```
Geant4Converter  INFO  +++  Successfully converted geometry to Geant4.

**************************************************** 
*  A runtime error has occured :                     
*    
-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : GeomMgt0002
      issued by : G4SmartVoxelHeader::BuildNodes()
PANIC! - Overlapping daughter with mother volume.
         Daughter physical volume tecmodule0r:TECModule0SideFrameLeft_1
         is entirely outside mother logical volume TECModule0 !!
-------- EEEE -------- G4Exception-END --------- EEEE -------

*  the program will have to be terminated - sorry.   
**************************************************** 
```

#### PR validation:

```
cmsRun DetectorDescription/DDCMS/test/python/testGeometry2021.py
```

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  std::vector<std::pair<G4LogicalVolume *, const cms::DDSpecPar *>> dd4hepVec;
+  const cms::DDSpecParRegistry &specPars = det->specpars();
+  cms::DDSpecParRefs specs;
+  specPars.filter(specs, "SensitiveDetector");
+  for (auto const &it : map_) {
+    for (auto const &fit : specs) {
+      for (auto const &pit : fit->paths) {
+        if (cms::dd::compareEqual(cms::dd::noNamespace(it.first.name()), cms::dd::realTopName(pit))) {
+          dd4hepVec.emplace_back(std::make_pair<G4LogicalVolume *, const cms::DDSpecPar *>(&*it.second, &*fit));
```
Comment (kpedro88) [inline SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc]: unnecessary to use make_pair w/ emplace_back

File: SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+      }
+    }
+  }
+
+  for (auto it : dd4hepVec) {
```
Comment (kpedro88) [inline SimG4Core/Geometry/src/DD4hep_DDG4Builder.cc]: use `const auto&`

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
@kpedro88 and @fabiocos - the unit test error is expected and is documented in the PR description:
```
Geant4Converter  INFO  +++  Successfully converted geometry to Geant4.

**************************************************** 
*  A runtime error has occured :                     
*    
-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : GeomMgt0002
      issued by : G4SmartVoxelHeader::BuildNodes()
PANIC! - Overlapping daughter with mother volume.
         Daughter physical volume tecmodule0r:TECModule0SideFrameLeft_1
         is entirely outside mother logical volume TECModule0 !!
-------- EEEE -------- G4Exception-END --------- EEEE -------

*  the program will have to be terminated - sorry.   
**************************************************** 

Failure using cmsRun /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-30-2300/src/SimG4Core/DD4hepGeometry/test/python/testZMM_13TeV_TuneCUETP8M1_cfi_GEN_SIM.py: status 1
status = 256

---> test DD4hepGeometryTestDriver had ERRORS

```

@smuzaffar - is there a way to allow this unit test failure without removing it from the tests? Thanks.

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
@smuzaffar - adding another PR to this one because I've asked the other one to be tested with this one is wrong. They do not depend on each other.

**Comment by ianna** [comment]
please abort tests

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
urgent

**Comment by kpedro88** [comment]
+upgrade

**Comment by ianna** [comment]
@civanch - please review

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Pull request merged**